exporter/signalfxexporter/config.go

```diff
@@ -37,10 +38,10 @@ var _ config.Unmarshallable = (*Config)(nil)
 
 // Config defines configuration for SignalFx exporter.
 type Config struct {
-	config.ExporterSettings        `mapstructure:",squash"`
-	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
-	exporterhelper.QueueSettings   `mapstructure:"sending_queue"`
-	exporterhelper.RetrySettings   `mapstructure:"retry_on_failure"`
+	config.ExporterSettings       `mapstructure:",squash"`
+	exporterhelper.QueueSettings  `mapstructure:"sending_queue"`
+	exporterhelper.RetrySettings  `mapstructure:"retry_on_failure"`
+	confighttp.HTTPClientSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
```

Comment:
I think we should delete the previous `Headers` field and rely on this one only since its squashed.  Operating on two maps throughout the exporter is unnecessary when we can just use the client settings one.

Reply:
Pushed the changes accordingly.